src/zarr/core/indexing.py

```diff
@@ -374,7 +376,7 @@ def __init__(self, dim_sel: slice, dim_len: int, dim_chunk_len: int) -> None:
 
     def __iter__(self) -> Iterator[ChunkDimProjection]:
         # figure out the range of chunks we need to visit
-        dim_chunk_ix_from = self.start // self.dim_chunk_len
+        dim_chunk_ix_from = 0 if self.start == 0 else self.start // self.dim_chunk_len
```

Comment:
```suggestion
        dim_chunk_ix_from = ceildiv(self.start, self.dim_chunk_len)
```

could this be used here?